Mathlib/Tactic/AdaptationNote.lean

```diff
@@ -0,0 +1,73 @@
+/-
+Copyright (c) 2024 Kyle Miller. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kyle Miller
+-/
+import Lean
+
+/-!
+# Adaptation notes
+
+This file defines a `#adaptation_note` command.
+Adaptation notes are comments that are used to indicate that a piece of code
+has been changed to accomodate a change in Lean core.
+They typically require further action/maintenance to be taken in the future.
+-/
+
+open Lean
+
+initialize registerTraceClass `adaptationNote
+
+/-- General function implementing adaptation notes. -/
+def reportAdaptationNote (f : Syntax → Meta.Tactic.TryThis.Suggestion) : MetaM Unit := do
+  let stx ← getRef
+  if let some doc := stx[1].getOptional? then
+    trace[adaptationNote] (Lean.TSyntax.getDocString ⟨doc⟩)
+  else
+    logError "Adaptation notes must be followed by a /-- comment -/"
+    let trailing := if let .original (trailing := s) .. := stx[0].getTailInfo then s else default
+    let doc : Syntax :=
+      Syntax.node2 .none ``Parser.Command.docComment (mkAtom "/--") (mkAtom "comment -/")
+    -- Optional: copy the original whitespace after the `#adaptation_note` token
+    -- to after the docstring comment
+    let doc := doc.updateTrailing trailing
+    let stx' := (← getRef)
+    let stx' := stx'.setArg 0 stx'[0].unsetTrailing
+    let stx' := stx'.setArg 1 (mkNullNode #[doc])
+    Meta.Tactic.TryThis.addSuggestion (← getRef) (f stx') (origSpan? := ← getRef)
+
+/-- Adaptation notes are comments that are used to indicate that a piece of code
+has been changed to accomodate a change in Lean core.
+They typically require further action/maintenance to be taken in the future. -/
+elab "#adaptation_note " (docComment)? : command => do
+  Elab.Command.liftTermElabM <| reportAdaptationNote (fun s => (⟨s⟩ : TSyntax `tactic))
+
+/-- Adaptation notes are comments that are used to indicate that a piece of code
+has been changed to accomodate a change in Lean core.
+They typically require further action/maintenance to be taken in the future. -/
```

Comment:
```suggestion
@[inherit_doc adaptationNoteCmd]
```
... you should be able to do this!

I might have gotten the syntax slightly wrong, this is untested!